functions/v2/imagemagick/main.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# [START functions_imagemagick_setup]
+import os
+import tempfile
+
+import functions_framework
+from google.cloud import storage, vision
+from wand.image import Image
+
+storage_client = storage.Client()
+vision_client = vision.ImageAnnotatorClient()
+# [END functions_imagemagick_setup]
+
+
+# [START functions_imagemagick_analyze]
+# Blurs uploaded images that are flagged as Adult or Violence.
+@functions_framework.cloud_event
+def blur_offensive_images(cloud_event):
+    file_data = cloud_event.data
+
+    file_name = file_data["name"]
+    bucket_name = file_data["bucket"]
+
+    blob = storage_client.bucket(bucket_name).get_blob(file_name)
+    blob_uri = f"gs://{bucket_name}/{file_name}"
+    blob_source = vision.Image(source=vision.ImageSource(gcs_image_uri=blob_uri))
+
+    # Ignore already-blurred files
+    if file_name.startswith("blurred-"):
+        print(f"The image {file_name} is already blurred.")
+        return
+
+    print(f"Analyzing {file_name}.")
+
+    result = vision_client.safe_search_detection(image=blob_source)
+    detected = result.safe_search_annotation
+
+    # Process image
+    if detected.adult == 5 or detected.violence == 5:
```

Comment:
Would a comment help to explain what the value being 5 or other values could mean? If it's self explanatory then fine to leave as is.

Reply:
+1 to add more detailed comment here

Reply:
Done.